task/mgmt-gateway/src/mgs_gimlet.rs

```diff
@@ -764,25 +770,28 @@ impl HostFlashUpdate {
             return Err(ResponseError::UpdateNotPrepared);
         }
 
-        match self.buf.ingest_chunk(
-            chunk.stream_id,
-            &self.task,
-            chunk.offset,
-            data,
-        )? {
-            UpdateProgress::Complete => {
-                // Update complete; we can now accept a new update.
-                self.buf.reset();
+        self.buf
+            .ingest_chunk(&chunk.id, &self.task, chunk.offset, data)
+    }
+
+    fn abort(&mut self, id: &UpdateId) -> Result<(), ResponseError> {
+        // We will allow the abort if either:
+        //
+        // 1. We have an in-progress update that matches `id`
+        // 2. We do not have an in-progress update
+        //
+        // We only want to return an error if we have a _different_ in-progress
+        // update.
+        match self.buf.in_progress_update_id() {
```

Comment:
Same suggestion as above (using a conditional + `matches!` instead of a `match`)